pkg/httpclient/json.go

```diff
@@ -119,7 +109,7 @@ func (c *Client) Hijack(method, path string, header http.Header, in interface{})
 	if err != nil {
 		return nil, err
 	}
-	dial := c.Dial
+	dial := c.HTTP.Transport.(*http.Transport).Dial
```

Comment:
I was a bit careful because of the `NewClientWithPin`, that specifies it's own dial in the http client:
https://github.com/flynn/flynn/blob/eea338421d545727fb6eb179d5e1bfd634f74676/controller/client/client.go#L68